doxylicious/README.md

```diff
@@ -0,0 +1,229 @@
+# Doxylicious
+
+## What is this?
+
+Kroxylicious has adopted [modular documentation](https://redhat-documentation.github.io/modular-docs/#) as a way of structuring content for end-users.
+[_Procedures_](https://redhat-documentation.github.io/modular-docs/#con-creating-procedure-modules_writing-mod-docs) are a particular kind of documentation module that explains how  to perform a specific task. 
+Procedures have a prescriptive structure, which includes:
+
+* listing prerequisite procedures,
+* the steps to be taken in this procedure,
+* optionally, steps to be taken to verify that this procedure worked.
+
+Procedures need to be tested to confirm that the documented steps actually work with (a release of) the software they're describing.
+
+This module enables defining a procedure in a form which can be used both
+* to generate the procedure documentation, in AsciiDoc 
+* to execute the procedure as a test, using the Junit 5 framework
+
+In order words, this makes it possible to automatically test some of your documentation, eliminating toil and the possibility of divergence between what's documented and what actually works.
+
+## Modules 
+
+* `doxylicious-core` defines all the core functionality.
+* `doxylicious-junit-provider` implements a Junit5 extension to enable execution of procedures as tests.
+* `doxylicious-maven-plugin` implements a Maven plugin for generating the procedure AsciiDocs.
+
+## Machine-readable procedures
+
+AsciiDoc is a fine format for authoring documentation, but was never intended to be executable. 
+These tools instead accept procedures defined in YAML (or JSON if you prefer) as a single source of truth for procedures.
+The AsciiDoc form of the procedure can be generated from the YAML.
+
+### A basic proc
+
+Let's start with a simple procedure
+
+```yaml
+id: succeeding_reliably # <1>
+title:
+  adoc: Succeeding reliably # <2>
+abstract:
+  adoc: 'This procedure is a minimal example.' # <3>
+intro:
+  adoc: |- # <4>
+    This procedure will always succeed because the only thing actually executed is the command `true`, which always returns 
+     with a 0 error code.
+procedure:
+  - step: # <5>
+      - adoc: Execute `true`.
+      - exec: true
+      - adoc: It is expect that this command returns without any output.
+  - step:
+      - adoc: Drink some coffee.
+verification:
+    - step: # <6>
+          - adoc: Check that coffee was drunk.
+          - exec: true
+```
+
+1. Each procedure must have a unique `id`.
+2. Published procedures¹ must have an AsciiDoc `title`.
+3. Published procedures may have an AsciiDoc `abstract`.
+4. Published procedures may have an AsciiDoc `intro`.
```

Comment:
I'm slightly confused as to why each of these nodes requires a child `adoc` node if by definition they are required to be AsciiDoc

Reply:
I wanted the structure of the YAML to reflect the mandated structure of a procedure. Sure we could make the YAML just have a single `header` or `prelude` property, where the user writes whatever sections they want. But that doesn't guide people into sticking with the structure. It's certainly an aesthetic choice, as having separate properties is not required in order to be able to execute it. 

I was also keen on having a JSON schema, for editing support in Intellij, where having the separate properties, each appropriately documented would have made for a much nicer authoring experience. However, I haven't been able to get intellij to play ball (at least not without a schema published on the web/to SchemaStore).